static/js/components/dashboard/courses/StatusMessages.js

```diff
@@ -133,8 +134,41 @@ export const calculateMessages = (props: CalculateMessagesProps) => {
     if (hasPaidForAnyCourseRun(course)) {
       // exam is required, user has not yet passed it
       if (exams && ! passedExam) {
+        let messageBox = {};
+        if (failedExam) {
```

Comment:
It enters the if on line 136 if a student doesn't have any passing exam grade or has not taken any exams yet. So the `if(failedExam)` gets executed only if the learner doesn't have a passing exam, meaning it could be empty list or only failed exams.